pkg/monitor/azure/nsg/nsg.go

```diff
@@ -0,0 +1,176 @@
+package nsg
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"net/netip"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2020-08-01/network"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/metrics"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/network"
+)
+
+const (
```

Comment:
some of these are used within the cluster monitor as additional dimmensions we pass when we construct the cluster monitor.  Can we pull these into a single place and make sure they're consistent?  

i.e. resourceGroup, subscriptionid, clusterResourceId

Reply:
https://github.com/Azure/ARO-RP/blob/22a6d6063c3c18ec3a1e39bcdc107c7b403ac9d5/pkg/monitor/cluster/cluster.go#L65-L69

Make sure the naming is consistent as well.

Reply:
+1, we should store the `dims` in a separate map, and then the codes/metrics in the const.